src/layout-grid.scss

```diff
@@ -2,44 +2,147 @@
 @import "./mixins";
 
 /*!
-.fd-layout-grid
+.fd-col
 */
 
-$block: #{$fd-namespace}-layout-grid;
+$block: #{$fd-namespace}-col;
+$blockContainer: #{$fd-namespace}-container;
+$blockRow: #{$fd-namespace}-row;
+
+.#{$blockContainer} {
+  @include fd-reset();
+
+  display: flex;
+  flex-wrap: wrap;
+  flex: 0 0 100%;
+  position: relative;
+  margin: 0 -0.25rem;
+
+  &--no-gap {
+    margin: 0;
+  }
+
+  > *:last-child {
+    margin-bottom: 0;
+  }
+
+  .#{$blockRow} {
+    margin-left: 0;
+    margin-right: 0;
+  }
+}
+
+.#{$blockRow} {
+  @include fd-reset();
+
+  display: flex;
+  flex-wrap: wrap;
+  flex: 0 0 100%;
+  position: relative;
+  margin: 0 -0.25rem 0.5rem -0.25rem;
+  min-width: 100%;
+
+  &--inner-flex {
+    display: flex;
+  }
+
+  .#{$block} {
+    .#{$blockRow} + .#{$blockRow} {
+      margin-top: 0.5rem;
+    }
+  }
+}
 
 .#{$block} {
   @include fd-reset();
 
-  display: grid;
-  grid-template-columns: repeat(3, 1fr);
-  grid-gap: var(--fd-width-gutter, $fd-width--gutter);
+  padding: 0 0.25rem;
+  min-width: 100%;
+  max-width: 100%;
+
+  @for $n from 1 through 12 {
+    &--#{$n} {
+      min-width: ( ( $n / 12 ) *100%);
+      max-width: ( ( $n / 12 ) *100%);
+    }
+  }
+
+  @for $n from 1 through 12 {
+    &--offset-#{$n} {
+      margin-left: ( ( $n / 12 ) *100%);
+    }
+  }
 
   &--no-gap {
-    grid-gap: 0;
+    padding: 0;
+  }
+
+  .#{$blockRow} {
+    margin: -0.25rem 0;
+  }
+}
+
+@media (min-width: 600px) {
+  .#{$blockContainer} {
+    margin: 0 -0.5rem;
+
+    .#{$blockRow} {
+      margin-left: 0;
+      margin-right: 0;
+    }
 
-    & > * {
-      box-shadow: none;
+    &--no-gap {
+      margin: 0;
     }
   }
 
-  &--gap-1 {
-    grid-gap: 1px;
+  .#{$blockRow} {
+    margin-bottom: 1rem;
 
-    & > * {
-      box-shadow: none;
+    .#{$block} {
+      .#{$blockRow} + .#{$blockRow} {
+        margin-top: 1rem;
+      }
     }
   }
 
-  @each $n in 1, 2, 3, 4, 5, 6 {
-    &--col-#{$n} {
-      grid-template-columns: repeat(#{$n}, 1fr);
+  .#{$block} {
+    padding: 0 0.5rem;
+
+    @for $n from 1 through 12 {
+      &--#{$n}--m {
+        min-width: ( ( $n / 12 ) *100%);
+        max-width: ( ( $n / 12 ) *100%);
+      }
+    }
+
+    &--no-gap {
+      padding: 0;
+    }
+
+    .#{$blockRow} {
+      margin: 0 -0.5rem;
     }
   }
+}
+
+@media (min-width: 1024px) {
+  .#{$block} {
+    @for $n from 1 through 12 {
```

Comment:
You can simplify this storing 12, 16 to variable, creates a map function which gets number and return letters `l, xl` etc.

Reply:
fixed